modules/siddhi-core/src/main/java/io/siddhi/core/query/processor/stream/window/BatchWindowProcessor.java

```diff
@@ -52,9 +56,16 @@
         namespace = "",
         description = "A window that holds an incoming events batch. When a new set of events arrives, the previously" +
                 " arrived old events will be expired. Batch window can be used to aggregate events that comes " +
-                "in batches.",
+                "in batches. If it has the parameter length specified, then batch window process the batch " +
+                "as several chunks.",
         parameters = {
-                // TODO: 7/7/18 Add parameter batch.size.max
+                @Parameter(name = "window.length",
+                        description = "The length of a chunk",
+                        type = {DataType.INT},
+                        optional = true,
+                        defaultValue = "If length value was not given it assign 0 as length and process " +
+                                "the whole batch as once"
+                ),
```

Comment:
Add param overloads

Reply:
Fixed in https://github.com/siddhi-io/siddhi/pull/1356/commits/ec35d4fe9fbce37b53b082e3f682e0d3adb6c660 and https://github.com/siddhi-io/siddhi/commit/f53a105b8b7ffc89dda063c8a18a2d32ef196bed